src/main/java/com/velox/sloan/cmo/workflows/kapalibrary/IndexBarcodeToSampleAutoAssigner.java

```diff
@@ -40,52 +41,64 @@ public IndexBarcodeToSampleAutoAssigner() {
 
     @Override
     protected boolean shouldRun() throws Throwable {
-        return activeTask.getTask().getTaskOptions().containsKey("AUTOASSIGN INDEX BARCODES") && !activeTask.getTask().getTaskOptions().containsKey("_INDEXES_AUTO_ASIGNED");
+        return (activeTask.getTask().getTaskOptions().containsKey("AUTOASSIGN INDEX BARCODES") && !activeTask.getTask().getTaskOptions().containsKey("_INDEXES_AUTO_ASIGNED"))
+                || (activeTask.getTask().getTaskOptions().containsKey("TCRseq-IGO"));
     }
 
     public PluginResult run() throws ServerException {
         autoHelper = new AutoIndexAssignmentHelper();
         try {
             List<DataRecord> attachedSamplesList = activeTask.getAttachedDataRecords("Sample", user);
             List<DataRecord> attachedIndexBarcodeRecords = activeTask.getAttachedDataRecords("IndexBarcode", user);
-            if (attachedIndexBarcodeRecords.isEmpty()) {
-                clientCallback.displayError(String.format("Could not find any IndexBarcode records attached to the TASK '%s'", activeTask.getTask().getTaskName()));
-                logError(String.format("Could not find any IndexBarcode records attached to the TASK '%s'", activeTask.getTask().getTaskName()));
-                return new PluginResult(false);
-            }
-            if (attachedSamplesList.isEmpty()) {
-                clientCallback.displayError("No Samples found attached to this task.");
-                logError("No Samples found attached to this task.");
-                return new PluginResult(false);
-            }
-            if (StringUtils.isBlank(activeTask.getTask().getTaskOptions().get("AUTOASSIGN INDEX BARCODES")) &&
-                    !activeTask.getTask().getTaskOptions().get("AUTOASSIGN INDEX BARCODES").contains("INDEX TYPE")) {
-                clientCallback.displayError("Task Option 'VALIDATE UNIQUE SAMPLE RECIPE' is missing valid value. Please double check. Valid values should be in format 'INDEX TYPE (IDT | TruSeq)'");
-                logError("Task Option 'VALIDATE UNIQUE SAMPLE RECIPE' is missing valid value. Please double check. Valid values should be in format 'INDEX TYPE (IDT | TruSeq)'");
-                return new PluginResult(false);
-            }
+            List<DataRecord> attachedTCRseqBarcodeRecords = activeTask.getAttachedDataRecords("TCRseq-IGO", user);
+            List<Object> uniquePlates = getAllUniquesPlates(attachedSamplesList);
```

Comment:
I think this would return all the ancestor plates of the sample. We just need the parent plate record, which is the container the samples being processed are on.